### DISCUSSION TITLE
Share your knowledge

### DISCUSSION THREAD

**Discussion by souyakuchan** [souyakuchan/LLM_DD_Challenge#1] [Announcements]
創薬のための LLM 活用テクニックの共有･議論の場です。
Upvote 獲得数の多い参加者に与える賞も設ける予定。

**Comment by ThePioneerJP** [comment]
とりあえずtweetがリンクなどの文字列を直コピペできないので時短用に。
https://cache-challenge.org/challenges/Finding-ligands-targeting-the-TKB-domain-of-CBLB

そこにおいてあるSDファイルは、LLM入力に適した形の情報とし表示させたいのならVS codeなどのエディターで閲覧することが可能なので、そのデータをコピペしてinputすることで、精度が上がる可能性がある。
（ちなみに900個全部入れるのは無理なので、オンラインのラインシャッフラーに入れて、そのデータを100行ずつサンプルとして入れる方法を私は取っている。これまで30個作らせたが、そのうち29個のsmilesは、少なくとも、動作はしていた）

さて、LLMの運用にあたっては、一般論としてではあるが、専門家（薬学、生化学など、関連しそうな分野の専門家）のロールをプレイさせること、より高精度な回答を出すことが知られている英語を使うことで、得られる情報の精度が高くなる可能性が大きい。

また、そのロールを与えた上で、キーワードを説明させると、そのことによっても精度が上がる可能性はある。
こちらを行う場合は、キーワード説明をgoogleなどでダブルチェックし、間違いがあれば早めに正しておくと、より理解を深めた状態での推論が可能になる。

生成をsmilesで行った場合、その理論上の可能性のチェックは以下で可能。
https://www.cheminfo.org/flavor/malaria/Utilities/SMILES_generator___checker/index.html

生成したもののうちの10個を入れてみたところ、とりあえず9個は理論上smilesとしての存在は可能で、出力するとこんな感じに。
![image](https://user-images.githubusercontent.com/116901962/229266518-cb7a6e83-689d-4cbe-8d4a-e1b96a6da8a6.png)

一方で、実在しているものをそのまま出しているのではないことの確認は、重複ラインチェックをすることで実行可能。
https://dedupelist.com/

提出前の備忘録として、取りまとめた結果を念のためチェックしたら、どうも3つは実在のものをそのまま出していた可能性がある…。
![image](https://user-images.githubusercontent.com/116901962/229273174-f322a53d-e101-49f1-9e38-715ff4c3c000.png)
